src/Pages/Profile/admin/AdminView.jsx

```diff
@@ -0,0 +1,355 @@
+      .then(result => {
+        if (result.status >= 200 && result.status < 300) {
+          setToggleSubmit(!toggleSubmit)
+        }
+      })
+      .catch(err => {
+        window.alert('The Email has already existed, leave it blank if you do not want to change')
+        console.log(err)
+      })
+  }
+
+  /*
+  translating access Level into role
+  The access level is defined as follows:
+  -2: Ban
+  -1: Pending
+   0: Member
+   1: Officer
+   2: Admin
+  */
+  function roleTranslator(level) {
+    switch (level) {
+      case -2:
+        return 'Ban'
+      case 0:
+        return 'Member'
+      case 1:
+        return 'Officer'
+      case 2:
+        return 'Admin'
+      default:
+        return 'Pending'
+    }
+  }
+
+  function editModalButton() {
+    let FormGroups = [
+      {
+        label: 'First Name',
+        type: 'email',
+        placeholder: user.firstName,
+        onChange: (e) => setFirstName(e.target.value)
+      },
+      {
+        label: 'Last Name',
+        type: 'email',
+        placeholder: user.lastName,
+        onChange: (e) => setLastName(e.target.value)
+      },
+      {
+        label: 'Middle Initial',
+        type: 'email',
+        placeholder: user.middleInitial,
+        onChange: (e) => setMiddleInitial(e.target.value)
+      },
+      {
+        label: 'Email',
+        type: 'email',
+        placeholder: user.email,
+        onChange: (e) => setEmail(e.target.value)
+      },
+      {
+        label: 'Password',
+        type: 'password',
+        placeholder: 'make it secure',
+        onChange: (e) => setPassword(e.target.value)
+      },
+      {
+        label: 'Door Code',
+        type: 'email',
+        placeholder: 'make it secure',
+        onChange: (e) => setDoorCode(e.target.value)
+      }
+    ]
+
+    /*
+    translating access Level into role
+    The access level is defined as follows:
+    -2: Ban
+    -1: Pending
+     0: Member
+     1: Officer
+     2: Admin
+    */
+    let memberships = [0, 1, -1, 2, -2]
+
+    return (
+      <div>
+        <div>
+          <Button
+            color="primary"
+            style={{
+              position: 'relative',
+              left: '80%'
+            }}
+            onClick={() => { handleToggle() }}>
+            Edit
+          </Button>
+        </div>
+
+        <Modal isOpen={toggle}>
+          <ModalHeader>Modal title</ModalHeader>
+          <ModalBody>
+            <Form>
+
+              {FormGroups.map((group, index) => {
+                return <FormGroup key={index}>
+                  <Label>{group.label}</Label>
+                  <Input
+                    type={group.type}
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              {/*Need Improvements, not doing anything currently*/}
```

Comment:
Remove the commented out code below. you can open an issue to add photo upload and paste the code there